src/test/unit/callbacks/chained_writer_test.cpp

```diff
@@ -0,0 +1,113 @@
+#include <stan/callbacks/chained_writer.hpp>
+#include <gtest/gtest.h>
+
+namespace test {
+  class mock_writer : public stan::callbacks::writer {
```

Comment:
I did write a generic class that could get used instead of making one of these every time but... this is fine too. The tests just need to go, they don't need to be pretty.